mlflow/models/evaluation/base.py

```diff
@@ -1643,14 +1643,32 @@ def pred_sample(eval_df, _builtin_metrics, _artifacts_dir):
 
     _EnvManager.validate(env_manager)
 
+    # If Dataset is provided, the targets and predictions can only be specified by the Dataset,
+    # not the targets and predictions parameters of the mlflow.evaluate() API.
+    if isinstance(data, Dataset) and targets is not None:
+        raise MlflowException(
+            message="The top-level targets parameter should not be specified since a Dataset "
+            "is used. Please only specify the targets column name in the Dataset. "
+            "Meanwhile, please specify `mlflow.evaluate(..., targets=None, ...)`.",
+            error_code=INVALID_PARAMETER_VALUE,
+        )
+    if isinstance(data, Dataset) and predictions is not None:
```

Comment:
Can we check if the dataset specifies `predictions`? If it does, can we adjust the error message accordingly? I.e. if dataset doesn't have predictions, tell users how to set them (give an example). If the dataset does have predictions, tell them to remove the predictions argument from `mlflow.evaluate()`.

Reply:
ditto

Reply:
Looks great!